skimage/filters/thresholding.py

```diff
@@ -137,6 +137,9 @@ def try_all_threshold(image, figsize=(8, 5), verbose=True):
 
     Examples
     --------
+    .. testsetup::
```

Comment:
Any way to get these out of the docstrings and into a configuration file, e.g., if we used doctest++?

Reply:
doctest-plus doesn't have an [option](https://github.com/scientific-python/pytest-doctestplus?tab=readme-ov-file#skipping-tests) to do so in a configuration file, I think.

See https://github.com/scikit-image/scikit-image/pull/7289#discussion_r1448907293 for background on the current pattern. Like @soupault I actually like the explicitness of this and didn't like that the other option using `__doctest_skip__` is hiding that doctests are skipped.

Note that these directives don't show up in our rendered HTML docstrings. 

What do you think?

Reply:
You could add something like
```
needs_matplotlib = [
    "draw/draw.py",
    "filters/thresholding.py",
    ...
]

collect_ignore = []
try:
    import matplotlib

except ImportError:
    collect_ignore += needs_matplotlib
```
to `skimage/conftest.py`.

Reply:
It would be nice not to have the docstrings littered with statements that make little sense to new readers (for whom these are most useful).

Reply:
Hmm, I see the value in that. `.. testsetup` might appear often enough via `help(some_func)` and using IPython's `?` that it's a relevant concern. 

I'd support an alternative solution if it makes it visible that a test is skipped. Otherwise it might get really confusing for contributors. Imagine them not having installed an optional dependency and wondering why the CI is showing red for something they can't reproduce because the doctest is just silently skipped by some `collect_ignore` somewhere they have no idea about.

I can't get the solution suggested by @jarrodmillman to work reliably, it seems like doctests are still collected. Furthermore, it and [`collect_ignore_glob`](https://docs.pytest.org/en/7.1.x/reference/reference.html#globalvar-collect_ignore_glob) can only ignore entire files or directories.

Reply:
We've been using it for NetworkX reliably for several years. Can you describe how it is unreliable?

It does ignore entire files or directories. That hasn't been an issue for NetworkX.

Reply:
> Can you describe how it is unreliable?

Trying that out locally just doesn't seem to work. E.g. if I add `"restoration/j_invariant.py",` to https://github.com/scikit-image/scikit-image/blob/9107d523a899b8d3767630941ac28f438bf875d2/skimage/conftest.py#L5-L6

and run
```
spin test -- -v --doctest-plus skimage/restoration/j_invariant.py
```
it's still picked up. I'm guessing it's because I am somehow using the wrong pattern? I couldn't figure out one that works to skip the module.

Also in case of `skimage/restoration/j_invariant.py`  only the doctest of `denoise_invariant` needs to be skipped if `pywt` is not available. Skipping the entire module would skip two doctests unnecessarily. 

If we can make skipping in doctestplus more visible, then I think we can satisfy everyone. I'm currently looking into it in https://github.com/scientific-python/pytest-doctestplus/issues/246...